src/webgl/p5.RendererGL.js

```diff
@@ -365,6 +365,12 @@ p5.prototype.setAttributes = function(key, value) {
   this.push();
   this._renderer._resetContext(attr);
   this.pop();
+  if (
+    this._renderer._curCamera._renderer !== this._renderer &&
+    this._renderer._curCamera.cameraType === 'default'
```

Comment:
i'm not sure the `cameraType` check is necessary here.

i think this check can be as simple as
```javascript
if (this._renderer._curCamera) {
  this._renderer._curCamera._renderer = this._renderer;
}
```

Reply:
The idea behind `cameraType` check was taking precautions rather than waiting for a bug to appear but at this moment, I was not able to think of a situation where this could go wrong.

For the second suggestion, I think that could work for sure, didn't think of that while implementing. I think the current code is better in terms of readability, but anyways I will add a comment .

Reply:
i can understand the precaution, but it seems to me that the bug here is that the camera's `_renderer` property is wrong (once the renderer has been switched out), and that this is the case regardless of the `cameraType`. At this point in the code, the old renderer has been discarded - it's no longer attached to the canvas.

Honestly i'd rather that the camera didn't hold onto the renderer at all: the back reference is a design smell and this kind of state-related bug is why.